### PR TITLE
Adds message about data retention on gdpr overview

### LINKED ISSUE #12864: In the GDPR Overview display the log data retention policy as configured
Currently Super Users can decide to [delete old log data](https://matomo.org/docs/managing-your-databases-size/#deleting-old-unprocessed-data) "Raw data is deleted after X months", but currently it is not very clear to anyone what log data retention policy is in place. So we need to make this information clearly available to Super Users and admin users.

For example: in the GDPR Overview page (visible to Super Users and admins), we could display `Data retention: all visitor and actions raw data are deleted after 6 months.`

(and if the value is less than 1 month it would say "14 days"). 

Thoughts?

### PR DESCRIPTION
**Pull request by sgiehl**
fixes #12864

### CODE REVIEW

**Comment by tsteur** [comment]
FYI: We still have to mention the data retention period they should enter in a privacy notice, consent notice, or privacy policy. 

This would be usually whatever is higher - aggregated reports or raw data retention rate. If the delete reports delete all reports except the numeric reports, then we only show the raw/log data retention rate to be used for this. That's because aggregated reports still contain personal information and therefore need to be included in the data retention rate.

**Comment by mattab** [comment]
Maybe we should add a sentence below the bullet points "Data retention policy <a>can be set</a> by a Super User."  with the link only there for Super Users.
(IIRC the page "Anonymise data" is not visible to admin users)


@tsteur I would only mention both and let users pick the one they need depending on their needs and requirements. we can't  really go into more details for them, because they may not have PII (user id, custom dim) in their aggregate reports and then would only need to mention the raw data retention.

**Comment by tsteur** [comment]
> @tsteur I would only mention both and let users pick the one they need depending on their needs and requirements. we can't really go into more details for them, because they may not have PII (user id, custom dim) in their aggregate reports and then would only need to mention the raw data retention.

the thing be to at least educate them on helping to make a decision. Nobody can realistically know which one to pick so we need to help our users.

**Comment by sgiehl** [comment]
I'm fine with changing it again, just let me know what exactly to change/add.

**Comment by tsteur** [comment]
We would only need to add a message. Like "The overall data retention rate for your privacy policy is the raw data retention rate. Please note that aggregated reports may contain personal data as well. If you are using features like User ID, Custom Variables, Custom Dimension, or track personal data through other events such as events, page URLs or page titles, etc, then the overall data retention rate is the data retention rate that is higher. "

In reality, people sometimes track different websites and for some pages, they may have personal data in aggregated reports and for some not.

btw: I don't know if you use "Log data" somewhere, but lets maybe always use raw data instead as this is better understandable for people that don't know our tables etc are called "log" and also in general?

**Comment by sgiehl** [comment]
we use `log data` in some other translations in PrivacyManager. Should we replace it with `raw data` at all occurrences?

**Comment by tsteur** [comment]
Makes sense I would say @mattab ?

**Comment by mattab** [comment]
Sounds good to replace wording by raw data :+1:

File: plugins/PrivacyManager/templates/gdprOverview.twig
Comment (diosmosis) [inline plugins/PrivacyManager/templates/gdprOverview.twig]: Think it would be clearer here to write "the higher of the two" instead of "the data retention rate that is higher".

File: plugins/PrivacyManager/templates/gdprOverview.twig
Comment (diosmosis) [inline plugins/PrivacyManager/templates/gdprOverview.twig]: Typo, should be `aggregated reports`

File: plugins/PrivacyManager/templates/gdprOverview.twig
Comment (diosmosis) [inline plugins/PrivacyManager/templates/gdprOverview.twig]: Think it should be "visit and action raw data", since you can't technically delete the visitor themselves :P

File: plugins/PrivacyManager/templates/gdprOverview.twig
Comment (diosmosis) [reply plugins/PrivacyManager/templates/gdprOverview.twig]: Another suggestion: "through other events" => "in other ways"

File: plugins/PrivacyManager/Controller.php
Comment (diosmosis) [inline plugins/PrivacyManager/Controller.php]: Not sure how beneficial it is, but it would look a bit better to use the singular translation if the year/day/month is `1`. (ie, `1 year 1 month` instead of `1 years 1 months`)

**Comment by diosmosis** [comment]
Noticed a bug, when I put 17 months into the "delete old report data" section:

image

It shows up as 17 days in the GDPR section:

image

**Comment by diosmosis** [comment]
Not sure why but I'm getting weird results when I test this. If I set the "Delete old visitor logs" entry to 180 days & don't delete report data:

image

image

It says it will never delete raw data & will delete report data if > 1 year 5 months:

image

I think for the aggregated data section at least, it's not checking if the report deletion feature is enabled?

**Comment by diosmosis** [comment]
If both features are enabled:

image

image

then the gdpr overview says it will delete raw data, but it gets the number wrong (13 days instead of 182, I think it's using the wrong setting):

image

**Comment by sgiehl** [comment]
@diosmosis I've updated the PR with your suggestions and fixed some stuff.

**Review by diosmosis — APPROVED** [review]
Good to go 👍 . Not sure if @tsteur or @mattab has to do a final approval since it's GDPR related, so just approving.

**Comment by diosmosis** [comment]
Oh forgot about the UI tests, they will need to be updated before merging.

**Comment by mattab** [comment]
Feedback

* Could you please add a UI test which shows both bullet point values?

Maybe @tsteur you have some more feedback?

**Comment by sgiehl** [comment]
@mattab I've added a UI test. See https://builds-artifacts.matomo.org/matomo-org/matomo/gdprretention/27727/PrivacyManager_gdpr_overview_no_retention.png and https://builds-artifacts.matomo.org/matomo-org/matomo/gdprretention/27727/PrivacyManager_gdpr_overview.png

**Comment by tsteur** [comment]
I would change "then the overall data retention rate" => " then the overall data retention rate for your privacy policy ". 

For some reason I would have personally listed the raw data retention before the aggregated report but that is totally fine either way.

I don't know if we already have an FAQ, but if so it would be great to link raw data and aggregated data to an FAQ entry so it is clear what they mean. All good though if we don't have them.

**Comment by sgiehl** [comment]
changed the order and updated the text.
Regarding FAQ: I didn't find some matching entries, but maybe @mattab knows if there are any.

File: plugins/PrivacyManager/templates/gdprOverview.twig
Comment (diosmosis) [inline plugins/PrivacyManager/templates/gdprOverview.twig]: I just noticed, but are these variables named correctly? The `deleteReportsEnable` block talks about raw data & the `deleteLogsEnable` block talks about aggregated reports.

Comment (sgiehl) [reply plugins/PrivacyManager/templates/gdprOverview.twig]: seems I somehow screwed it up :see_no_evil:

**Pull request merged**
